### PR TITLE
fix(pipeline): reject illegal pipeline facade config

### LINKED ISSUE #437: Pipeline facade float32 overflow hardening and int32 bounds on dimensions and smoke probes
## Problem

End-to-end pipeline component configs (`Step2FeatureConfig`, `Step2UPGDConfig`, `Step2AssociativePipelineConfig`, `HordeActorCriticPipelineConfig`) and `run_pipeline_smoke` lack float32 overflow validation on scalar hyperparameters (`ema_decay`, `periods`, `step_size`, `sparsity`, `write_lr`, `retention`, `utility_lr`, `utility_decay`, `min_weight`, `max_weight`, `logit_scale`, `scope_lr`, `budget_lr`, `initial_budget_fraction`, `scope_logit_clip`, `actor_step_size`, `actor_lamda`, `temperature`, `actor_obgd_kappa`), allow adversarial float subclasses with out-of-domain exact ratios, and lack signed int32 bounds on `observation_dim`, `n_heads`, `hidden_sizes`, `vocab_size`, `block_size`, `suffix_length`, `max_features`, `min_effective_budget`, `n_actions`, `value_head_index`, and `run_pipeline_smoke` arguments (`steps`, `seed`).

## Reproduction

```python
from alberta_framework.pipeline import Step2UPGDConfig

# Accepts float32 overflow on step_size
cfg = Step2UPGDConfig(step_size=1e100)
```

## Proposed Fix

- Use `round_real_to_float32_with_ratio` in `alberta_framework/pipeline.py` checking exact integer ratios, domain bounds, and rejecting class-property spoofing.
- Bound integer dimensions, counts, sizes, and smoke arguments to signed int32 bounds (`2**31 - 1`).
- Add comprehensive test coverage in `tests/test_pipeline.py`.

### PR DESCRIPTION
**Pull request by deepanshu-yd**
Closes #437.

## What changed

`Step2FeatureConfig`, `Step2UPGDConfig`, `Step2AssociativePipelineConfig`, `HordeActorCriticPipelineConfig`, and `run_pipeline_smoke` now strictly validate scalar hyperparameters, float32 execution sink bounds, and exact integer ratios before constructing pipeline components. Float32 execution values use the shared exact-ratio `round_real_to_float32_with_ratio` helper, preventing binary64 double rounding and closing host-vs-ratio escapes for float subclasses.

Exact float32 overflow is rejected across `ema_decay`, `periods`, `step_size`, `sparsity`, `write_lr`, `retention`, `utility_lr`, `utility_decay`, `min_weight`, `max_weight`, `logit_scale`, `scope_lr`, `budget_lr`, `initial_budget_fraction`, `scope_logit_clip`, `actor_step_size`, `actor_lamda`, `temperature`, and `actor_obgd_kappa`. Integer dimensions, counts, sizes, and smoke arguments (`steps`, `seed`) are bounded to standard signed int32 bounds (`2**31 - 1`).

## Scope

• `alberta_framework/_float32.py`
• `alberta_framework/pipeline.py`
• `tests/test_pipeline.py`

No registered/frozen source, `outputs/**`, protocol, seed, changelog, or evidence artifact changed.

## Exact-head verification

• Pipeline test suite: 150 passed in 23s;
• full Ruff: passed;
• strict mypy: no issues in 164 source files;
• `git diff --check`: clean.

## Scientific integrity

This is facade input validation, not scientific evidence or a performance claim.

## Contribution provenance

• AI assistance: yes
• AI provider/model: google / gemini-3.7-flash
• Client / agent tooling: Antigravity CLI
• Attribution status: self-reported

### CODE REVIEW

**Review by lalalune — CHANGES_REQUESTED** [review]
Blocking review at exact `16acc31e67bd9eb3e7f053cf364eda7e2b69c769`:

1. This head rewrites `alberta_framework/_float32.py` from an obsolete base and now conflicts with current main. Its `_real_ratio` reintroduces the pre-#390 `isinstance(value, Integral)`/hook surface. Rebase and consume the repaired shared helper; do not overwrite it.
2. Pipeline `_require_int` accepts actual integer subclasses and invokes `__int__`: `Step2UPGDConfig(observation_dim=LieInt(-1))` was accepted/canonicalized as 4. Reject subclasses before hooks while retaining exact builtin/NumPy integer types.
3. Bool flags use spoofable `isinstance`. Objects reporting `__class__ = bool` were retained by `use_layer_norm` and `normalize_by_weight`; both configs then failed JSON serialization. Require actual bool and canonicalize/store it.
4. String discriminators trust equality. An object equal only to `"default"` was retained as `learner_preset`, constructed successfully, then failed JSON serialization. Require actual supported strings for every preset/mode/family discriminator.

The ordinary float32/int32 cases are directionally correct, but the facade is not fail-closed or serializable yet. Please add direct/config-roundtrip regressions for these cases and rerun on current main.
